app/models/customer.py

```diff
@@ -25,10 +25,10 @@ def __repr__(self):
 class Telephone(db.Model):
     __tablename__ = "telephone"
     id: Mapped[int] = mapped_column(Integer, primary_key=True)
-    phone_number: Mapped[str] = mapped_column(String(255))
+    phone_number: Mapped[Optional[str]] = mapped_column(String(255))
     customer_id: Mapped[int] = mapped_column(ForeignKey("customer.id", ondelete="CASCADE"))
 
-    customer: Mapped["Customer"] = relationship("Customer", back_populates="phone_numbers")
+    customer: Mapped[Optional["Customer"]] = relationship("Customer", back_populates="phone_numbers")
```

Comment:
I think we need to require at least one phone number.  What do you think?

Reply:
It still can't be null on the Customer side of the relationship. Now that you point that out though, having the phone number nullable in the Telephone model is no longer necessary. When I was originally writing this I was removing the phone number from the session first and then deleting it, and doing it that way it had to be nullable in the Telephone model. Pushing another commit now to remove the optional from the models and rebasing.